src/Microsoft.Azure.Functions.Extensions.Mcp/Abstractions/IResourceRegistry.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using ModelContextProtocol.Protocol;
+
+namespace Microsoft.Azure.Functions.Extensions.Mcp.Abstractions;
+
+/// <summary>
+/// Represents a registry for MCP resources.
+/// </summary>
+internal interface IResourceRegistry
+{
+    /// <summary>
+    /// Registers an MCP resource.
+    /// </summary>
+    /// <param name="resource"></param>
+    void Register(IMcpResource resource);
+
+    /// <summary>
+    /// Tries to get an MCP resource by its URI.
+    /// </summary>
+    /// <param name="uri">The URI of the resource.</param>
+    /// <param name="resource">The retrieved resource, if found.</param>
+    /// <returns>True if the resource was found; otherwise, false.</returns>
+    bool TryGetResource(string uri, [NotNullWhen(true)] out IMcpResource? resource);
+
+    /// <summary>
+    /// Gets all registered MCP resources.
+    /// </summary>
+    ICollection<IMcpResource> GetResources();
```

Comment:
Should probably be a read-only type, so no caller expects to mutate it.

```suggestion
    IReadOnlyCollection<IMcpResource> GetResources();
```

Also, what is the difference between `GetResources` and `ListResourcesAsync`?

Reply:
Having one be async and the other not gives a bit of weird ergonomics. They are directly related (as your implementation shows, one is a projection of another). But the one being non-async effectively makes the other one non-async as well in real use scenarios.